src/Interpreters/executeQuery.cpp

```diff
@@ -1766,7 +1766,8 @@ std::pair<ASTPtr, BlockIO> executeQuery(
     ASTPtr ast;
     BlockIO res;
     auto implicit_tcl_executor = std::make_shared<ImplicitTransactionControlExecutor>();
-    res = executeQueryImpl(query.data(), query.data() + query.size(), context, flags, stage, nullptr, ast, implicit_tcl_executor);
+    ReadBufferUniquePtr no_input_buffer;
```

Comment:
not needed

Reply:
we need to pass it by reference to executeQueryImpl.